### PR TITLE
[MOB-19204] Provide an API to update inbound content in the SDK

### PR DESCRIPTION
**Pull request by swarna04**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPMessaging/Sources/Event+Messaging.swift [hunk 1]
```diff
@@ -261,16 +261,23 @@ extension Event {
 
     // MARK: - Update Feed Messages Public API Event
```
Comment (sbenedicadb) [inline AEPMessaging/Sources/Event+Messaging.swift:262]: nit: let's update this comment as well

File: AEPMessaging/Sources/Messaging+PublicAPI.swift [hunk 2]
```diff
@@ -70,24 +70,24 @@ import UserNotifications
...
-    /// Dispatches an event to fetch message feeds for the provided surface paths from the Adobe Journey Optimizer via the Experience Edge network.
-    /// - Parameter surfacePaths: An array of surface path strings.
-    static func updateFeedsForSurfacePaths(_ surfacePaths: [String]) {
-        let validSurfacePaths = surfacePaths
-            .filter { !$0.isEmpty }
+    /// Dispatches an event to fetch propositions for the provided surfaces from remote.
+    /// - Parameter surfaces: An array of surface objects.
+    static func updatePropositionsForSurfaces(_ surfaces: [Surface]) {
+        let validSurfaces = surfaces
+            .filter { !$0.isValid }
```
Comment (sbenedicadb) [inline AEPMessaging/Sources/Messaging+PublicAPI.swift:77]: should the evaluation here be `{ $0.isValid }` instead?

File: AEPMessaging/Sources/Messaging.swift [hunk 3]
```diff
@@ -171,33 +169,32 @@ public class Messaging: NSObject, Extension {
...
-        let requestedSurfaces = surfacePaths
-            .filter { !$0.isEmpty }
-            .map { appSurface + MessagingConstants.PATH_SEPARATOR + $0 }
-            .filter { isValidSurface($0) }
+        let requestedSurfaceUris = surfaces
+            .filter { $0.isValid }
+            .compactMap { $0.uri }
 
-        if requestedSurfaces.isEmpty {
+        if requestedSurfaceUris.isEmpty {
```
Comment (sbenedicadb) [inline AEPMessaging/Sources/Messaging.swift:188]: prefer `guard`

**Review by sbenedicadb — CHANGES_REQUESTED** [review]
a couple small changes

**Comment by swarna04** [comment]
@sbenedicadb I've incorporated feedback here. Thanks for the review!

**Review by sbenedicadb — APPROVED** [review]

**Pull request merged**